src/rules/date-rules.ts

```diff
@@ -9,10 +9,11 @@ export const DateRules = {
   }
 }
 
-// validation helper methods
+// validation helper method(s)
 export const DateRuleHelpers = {
   isBetweenDates (minDate: Date, maxDate: Date, dateStrToValidate: string): boolean {
     if (!dateStrToValidate) { return true }
-    return (new Date(dateStrToValidate) >= minDate) && (new Date(dateStrToValidate) <= maxDate)
+    const date = new Date(dateStrToValidate) // eg, September 5, 2022
```

Comment:
I don't like this one but there's nothing I can do about it (other than refactoring the shared date picker).

The problem is that the shared date picker returns this date like this: "September 5, 2022". But this statement converts it to a Date object according to the browser's local timezone, which may be incorrect or spoofed, so the date could be Sept 4 or Sept 6.

One possibility to solve this is to convert the string version of the date to a UTC date (using known tz offset). However, we can't just compare against an array of month names, since the locale could be a different language :(

A better solution would be for the shared date picker to return a date as YYYY-MM-DD. Even then, this code would be ugly as we can't use the Date Mixin to convert dates so we'd have to copy the needed date mixin methods locally.

So I think we just leave this. Worst case, the date picker rejects a user's date selection, which doesn't break anything on our end :/

Reply:
I ran into this same problem

Reply:
Probably a good idea, unless you think the number of people using the site from another TZ is very small - I wonder what the stats are?

Reply:
Well, some of BC is Mountain Time, and we probably have some users from across Canada (if not the world). I think it should be fixed, or at least kept track of.

Reply:
I changed my mind slightly. What we should do here is compare dates only (not date-times). I'll still create a new ticket for this.

Reply:
New ticket for this: https://app.zenhub.com/workspaces/entities-team-space-6143567664fb320019b81f39/issues/bcgov/entity/13531